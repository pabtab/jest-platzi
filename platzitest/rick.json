{
  "id": 1,
  "name": "Rick Aveiro",
  "status": "Alive",
  "species": "Human",
  "gender": "Male"
}