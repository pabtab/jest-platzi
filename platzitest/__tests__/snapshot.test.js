import { getCharacter } from '../snapshot';
import rick from '../rick.json';

describe('Es hora de las instantaneas', () => {
  test('Snapshot', () => {
    expect(getCharacter(rick)).toMatchSnapshot()
  })

  test('Siempre fallan las instantaneas', () => {
    const user = {
      createAt: new Date(),
      id: Math.floor(Math.random() * 20)
    }

    expect(user).toMatchSnapshot()
  })

  test('Tenemos una excepcion dentro del codigo', () => {
    const user = {
      id: Math.floor(Math.random() * 20),
      name: 'Pablo Tabares'
    }

    expect(user).toMatchSnapshot({ id: expect.any(Number) })
  })
  
  
  
})
