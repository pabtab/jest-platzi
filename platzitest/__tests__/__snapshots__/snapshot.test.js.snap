// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las instantaneas Siempre fallan las instantaneas 1`] = `
Object {
  "createAt": 2018-11-11T23:40:11.664Z,
  "id": 0,
}
`;

exports[`Es hora de las instantaneas Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Aveiro",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`Es hora de las instantaneas Tenemos una excepcion dentro del codigo 1`] = `
Object {
  "id": Any<Number>,
  "name": "Pablo Tabares",
}
`;
